plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -103,16 +103,30 @@ export async function createRouter<
         checks,
         entities,
       }: { checks: string[]; entities: CompoundEntityRef[] } = req.body;
+      let error = undefined;
       const tasks = entities.map(async entity => {
         const entityTriplet =
           typeof entity === 'string' ? entity : stringifyEntityRef(entity);
-        const results = await factChecker.runChecks(entityTriplet, checks);
-        return {
-          entity: entityTriplet,
-          results,
-        };
+        try {
+          const results = await factChecker.runChecks(entityTriplet, checks);
+          return {
+            entity: entityTriplet,
+            results,
+          };
+        } catch (e: any) {
+          const errorMessage = `Failed to run check for entity ${entityTriplet} due to error: ${e.message}`;
+          logger.error(errorMessage);
+          error = errorMessage;
+          return {
+            entity: entityTriplet,
+            results: [],
+          };
+        }
       });
       const results = await Promise.all(tasks);
+      if (error) {
+        return res.status(500).send({ error, results });
```

Comment:
I'd say a 500 is an actual failure of the server itself, like if something threw an exception in an unexpected way and you wanted to almost raise an alarm.

Should we design this response differently? Should it be the case that when you run checks, you get back a complete list of checks every time but some of them might have a status of "couldn't complete, because of error X happening". That way you can always run the checks, and enumerate them in the view, and draw either a green or yellow or an X or whatever, to signify that you don't have data for this one.

Look for example at what the build output is in github itself on a PR. Just because one of the actions failed doesn't mean that you can't see the actions list. You'll see that all of them are green, except one which failed.

Reply:
Yea good shout. I think I'll do that unless all fail in which case I'll return the 500.